packages/circuits/jwt-verifier-template.circom

```diff
@@ -0,0 +1,326 @@
+
+    // Assert that period exists at periodIndex
+    signal period <== ItemAtIndex(maxMessageLength)(message, periodIndex);
+    period === 46;
+
+    // Assert that period is unique
+    signal periodCount <== CountCharOccurrences(maxMessageLength)(message, 46);
+    periodCount === 1;
+
+    // Find the real message length
+    signal realMessageLength <== FindRealMessageLength(maxMessageLength)(message);
+
+    // Calculate the length of the Base64 encoded header and payload
+    signal b64HeaderLength <== periodIndex;
+    signal b64PayloadLength <== realMessageLength - b64HeaderLength - 1;
+
+    // Extract the Base64 encoded header and payload from the message
+    signal b64Header[maxB64HeaderLength] <== SelectSubArrayBase64(maxMessageLength, maxB64HeaderLength)(message, 0, b64HeaderLength);
+    signal b64Payload[maxB64PayloadLength] <== SelectSubArrayBase64(maxMessageLength, maxB64PayloadLength)(message, b64HeaderLength + 1, b64PayloadLength);
+
+    // Calculate the maximum length of the decoded header and payload
+    var maxHeaderLength = (maxB64HeaderLength * 3) \ 4;
+    var maxPayloadLength = (maxB64PayloadLength * 3) \ 4;
+
+    // Decode the Base64 encoded header and payload
+    signal header[maxHeaderLength] <== Base64Decode(maxHeaderLength)(b64Header);
+    signal payload[maxPayloadLength] <== Base64Decode(maxPayloadLength)(b64Payload);
+
+    // Verify if the typ in the header is "JWT"
+    var typLength = JWT_TYP_LENGTH();
+    var typ[typLength] = JWT_TYP();
+    signal typMatch[typLength] <== RevealSubstring(maxHeaderLength, typLength, 1)(header, jwtTypStartIndex, typLength);
+    for (var i = 0; i < typLength; i++) {
+        typMatch[i] === typ[i];
+    }
+
+    // Verify if the key `kid` in the header is unique
+    var kidKeyLength = JWT_KID_KEY_LENGTH();
+    var kidKey[kidKeyLength] = JWT_KID_KEY();
+    signal kidKeyMatch[kidKeyLength] <== RevealSubstring(maxHeaderLength, kidKeyLength, 1)(header, jwtKidStartIndex, kidKeyLength);
+    for (var i = 0; i < kidKeyLength; i++) {
+        kidKeyMatch[i] === kidKey[i];
+    }
+
+    // Reveal the kid in the header
+    signal kidStartIndex <== jwtKidStartIndex + kidKeyLength + 1;
+    var kidLength = JWT_KID_LENGTH();
+    signal kidBytes[kidLength] <== RevealSubstring(maxHeaderLength, kidLength, 0)(header, kidStartIndex, kidLength);
+    kid <== Hex2FieldModular(kidLength)(kidBytes);
+
+    // Verify if the key `iss` in the payload is unique
+    var issKeyLength = ISS_KEY_LENGTH();
+    var issKey[issKeyLength] = ISS_KEY();
+    signal issKeyMatch[issKeyLength] <== RevealSubstring(maxPayloadLength, issKeyLength, 1)(payload, issKeyStartIndex, issKeyLength);
+    for (var i = 0; i < issKeyLength; i++) {
+        issKeyMatch[i] === issKey[i];
+    }   
+
+    // Reveal the iss in the payload
+    signal issStartIndex <== issKeyStartIndex + issKeyLength + 1;
+    signal issMatch[issLen] <== RevealSubstring(maxPayloadLength, issLen, 0)(payload, issStartIndex, issLength);
```

Comment:
Is it secure to skip the uniqueness check here?

Reply:
Same as the `kid` check.